### PR TITLE
add module to make noise reproduceably

### PR DESCRIPTION
**Pull request by ahnitz**
Add a module to make noise that is the same at a given time and seed value. This makes it independent of the start / end of the noise you request. If you give the same seed value, the same time will have the same data. This makes it a lot easier to coordinate workflows on fake data without having to pregenerate  frame files, or in other programs where fake noise is needed with different overlapping intervals.

### CODE REVIEW

File: pycbc/noise/reproduceable.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        The seed to generate the noise.
+
+    Returns
+    --------
+    noise : TimeSeries
+        A TimeSeries containing gaussian noise
+    """
+    # This is reproduceable because we used fixed seeds from known values
+    s = int(start / BLOCK_SIZE)
+    e = int(end / BLOCK_SIZE)
```
Comment (spxiwh) [inline pycbc/noise/reproduceable.py]: In the case that end / BLOCK_SIZE = end // BLOCK_SIZE you'll get 100s of data extra.

File: pycbc/noise/reproduceable.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    sv = RandomState(seed).randint(-2**50, 2**50)
+    data = numpy.concatenate([block(i + sv) for i in numpy.arange(s, e + 1, 1)])
```
Comment (spxiwh) [inline pycbc/noise/reproduceable.py]: At this point should you not throw away the extra data that is not needed? Ie if asking for 250s of data I'll return 300s.

File: pycbc/noise/reproduceable.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    Returns
+    --------
+    noise : TimeSeries
+        A TimeSeries containing gaussian noise colored by the given psd.
+    """ 
+    white_noise = normal(start_time - FILTER_LENGTH, end_time + FILTER_LENGTH,
+                          seed=seed)   
+    flen = (SAMPLE_RATE / psd.delta_f) / 2 + 1
+    psd.resize(flen)
```
Comment (spxiwh) [inline pycbc/noise/reproduceable.py]: This changes an input vector, which might not be obvious to a user. Shouldn't this take a view of the PSD? (ie. don't copy memory, but don't change the input object in place)

Comment (spxiwh) [inline pycbc/noise/reproduceable.py]: (that would be a useful method of the Frequency/Time Series classes if it doesn't already exist!)

**Comment by spxiwh** [comment]
And fix the whitespace.

File: pycbc/noise/reproduceable.py (hunk 1 above)
Comment (ahnitz) [inline pycbc/noise/reproduceable.py]: I assume you mean you'll get one extra block of 100s. Yeah, that's right. I"ll address your other issues first.

File: pycbc/noise/reproduceable.py (hunk 3 above)
Comment (ahnitz) [inline pycbc/noise/reproduceable.py]: @spxiwh What would be a useful method? A write protected view?

File: pycbc/noise/reproduceable.py (hunk 1 above)
Comment (ahnitz) [inline pycbc/noise/reproduceable.py]: I decided to add a separate line to handle this. I couldn't think of a clearer way.

File: pycbc/noise/reproduceable.py (hunk 3 above)
Comment (spxiwh) [inline pycbc/noise/reproduceable.py]: Yeah, maybe. It just seems wasteful to duplicate memory here ... But probably it doesn't matter overall.

Comment (ahnitz) [inline pycbc/noise/reproduceable.py]: In this case, I need to make it longer to change the upper frequency cutoff. I don't know how to do that *without* copying to new memory somewhere.

Comment (ahnitz) [inline pycbc/noise/reproduceable.py]: In the case of making it shorter, I agree a view would be fine, but I didn't add a case checking for which case we were in. I doubt most people are going to pass a PSD that goes beyond 8KHz though, so it wouldn't happen often that you'd get memory savings.

Comment (spxiwh) [inline pycbc/noise/reproduceable.py]: Seems a little weird though to generate noise at 16384Hz with a PSD that doesn't go that far??

**Comment by spxiwh** [comment]
Approved to merge, but please fix the whitespace.

**Pull request merged**